### ISSUE TITLE
Using interceptors on multipart request

### ISSUE THREAD

**Issue by abdulkash** [CodingAleCR/http_interceptor#123]
When i use a logging interceptor on a multipart request it returns StreamedResponse and if i listen to this object i get "Bad state exception stream has already been listened to"

**Comment by mittal-sahab-sudo** [comment]
Any fix for this ? I am getting the same issue.

**Comment by CodingAleCR** [comment]
Hi, I'm looking into it, however I do this on my spare time, which is definitely not enough. Pull requests are always welcomed though. 😄

**Comment by CodingAleCR** [comment]
At the moment, the way that you can work with Multipart Requests would be to do something like in the example project. In there, you'll find[ a way to do requests without necessarily needing to listen to the stream.](https://github.com/CodingAleCR/http_interceptor/blob/f7f9649284a44de05badd11f5cbca0a2f67abd23/example/lib/multipart_app.dart#L134)

I'm still looking for a way to "clone" the stream and data into a new one without having the stream either closed or _listened_ to but it might take me a while to figure out a scalable solution.

**Comment by abdulkash** [comment]
Its the same implementation on the example code that I used but it still has the same issue

**Comment by CodingAleCR** [comment]
What I meant with the example is that you might not need to listen to the stream, or is there any specific reason that you want to do that?

**Comment by abdulkash** [comment]
When I don't listen to the stream the logged response is StreamResponse object

**Comment by CodingAleCR** [comment]
Yes, you could work with the `StreamedResponse` like in the Multipart Request example. You would [parse the StreamedResponse to a Response and do what you need with it. ](https://github.com/CodingAleCR/http_interceptor/blob/main/example/lib/multipart_app.dart#L118-L156)

**Comment by CodingAleCR** [comment]
At least if you don't depend on `listen`. If you do then I would ask for a bit more patience since I can't look at it at the moment. Sorry it takes too long to get things done!

**Comment by eldarkk** [comment]
Any progress with fixing issue ?
